test/distributed/optim/test_named_optimizer.py

```diff
@@ -291,6 +291,56 @@ def test_load_state_dict(self):
             assert_equal=True,
         )
 
+    def test_load_state_dict_conditional_training(self):
+        """Check that NamedOptimizer load_state_dict works under conditional training case."""
+        m = TestDummyModel()
+        named_optim_1 = _NamedOptimizer(
+            m.named_parameters(),
+            torch.optim.SGD,
+            [
+                {"params": m.net1.parameters()},
+                {"params": m.net3.parameters(), "lr": 1e-3},
+            ],
+            lr=1e-2,
+            momentum=0.9,
+        )
+
+        for _ in range(2):
```

Comment:
sorry I may miss something, but where is conditional training happening here?

Reply:
So I only set parameters from net1 and net3 into param groups for the optim_1, while for optim_2, it is updating all parameters. So, the loading from the state from optim_1 to optim_2 only changes for the corresponding parameters.